data-in-pipeline/infra/lambda_code/handler.py

```diff
@@ -0,0 +1,94 @@
+import json
+import logging
+import os
+from http import HTTPStatus
+from typing import Any
+
+import boto3
+import psycopg2
+from psycopg2 import sql
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_secret(arn: str) -> dict[str, Any]:
+    """Retrieve secret from AWS Secrets Manager.
+
+    :param arn: ARN of the secret to retrieve
+    :type arn: str
+    :return: Secret value as a dictionary
+    :rtype: dict[str, Any]
+    """
+    sm = boto3.client("secretsmanager")
+    return json.loads(sm.get_secret_value(SecretId=arn)["SecretString"])
+
+
+def _load_sql(path: str, identifiers: dict[str, str]):
+    """Load SQL file and safely inject identifiers using psycopg2.sql.
+
+    :param path: Path to the SQL file
+    :type path: str
+    :param identifiers: Dictionary mapping placeholder names to identifier
+        values. Should contain: DB_NAME, LOAD_DB_USER, APP_SCHEMA
+    :type identifiers: dict[str, str]
+    :return: SQL object with safely injected identifiers
+    :rtype: sql.SQL
+    """
+    with open(path, "r", encoding="utf-8") as f:
+        sql_template = f.read()
+
+    # For string comparison in DO block, we need a literal
+    # For role/database/schema names, we need identifiers
+    load_db_user = identifiers["LOAD_DB_USER"]
+    sql_placeholders = {
+        "LOAD_DB_USER": sql.Identifier(load_db_user),
+        "LOAD_DB_USER_LITERAL": sql.Literal(load_db_user),
+        "DB_NAME": sql.Identifier(identifiers["DB_NAME"]),
+        "APP_SCHEMA": sql.Identifier(identifiers["APP_SCHEMA"]),
+    }
+
+    # Use sql.SQL().format() to safely inject identifiers and literals
+    return sql.SQL(sql_template).format(**sql_placeholders)
+
+
+def handler(event, context):
+    host = os.environ["AURORA_WRITER_ENDPOINT"]
+    db = os.environ["DB_NAME"]
+    port = int(os.environ.get("DB_PORT", "5432"))
+    master_secret_arn = os.environ["ADMIN_SECRET_ARN"]
+    sql_path = os.environ.get("SQL_PATH", "/var/task/create_iam_user.sql")
+    load_user = os.environ.get("LOAD_DB_USER", "load_db_user")
+    schema = os.environ.get("APP_SCHEMA", "public")
```

Comment:
question: Should we make these explicitly required and fail if they aren't in fear of generating test users?